src/Microsoft.ML.TimeSeries/IidChangePointDetector.cs

```diff
@@ -139,9 +181,106 @@ public override void Save(ModelSaveContext ctx)
             base.Save(ctx);
         }
 
-        public IDataTransform ApplyToData(IHostEnvironment env, IDataView newSource)
+        // Factory method for SignatureLoadRowMapper.
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+    }
+
+    public sealed class IidChangePointEstimator : IEstimator<ITransformer>
+    {
+        private readonly IHost _host;
+        private readonly string _inputColumnName;
+        private readonly string _outputColumnName;
+        private readonly int _confidence;
+        private readonly int _changeHistoryLength;
+
+        public IidChangePointEstimator(
+            IHostEnvironment env,
+            int confidence,
+            int changeHistoryLength,
+            string input,
+            string output)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register(nameof(IidChangePointEstimator));
+
+            _host.CheckNonEmpty(input, nameof(input));
+            _host.CheckNonEmpty(output, nameof(output));
+
+            _confidence = confidence;
+            _changeHistoryLength = changeHistoryLength;
+            _inputColumnName = input;
+            _outputColumnName = output;
+        }
+
+        public ITransformer Fit(IDataView input)
+        {
+            _host.CheckValue(input, nameof(input));
+            return new IidChangePointDetector(_host,
+                new IidChangePointDetector.Arguments
+                {
+                    Confidence = _confidence,
+                    ChangeHistoryLength = _changeHistoryLength,
+                    Source = _inputColumnName,
+                    Name = _outputColumnName,
+                });
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            var metadata = new List<SchemaShape.Column>() {
+                new SchemaShape.Column(MetadataUtils.Kinds.SlotNames, SchemaShape.Column.VectorKind.Vector, TextType.Instance, false)
+            };
+            var resultDic = inputSchema.Columns.ToDictionary(x => x.Name);
+            resultDic[_outputColumnName] = new SchemaShape.Column(
+                _outputColumnName, SchemaShape.Column.VectorKind.Vector, NumberType.R8, false, new SchemaShape(metadata));
+
+            return new SchemaShape(resultDic.Values);
+        }
+    }
+
+    public static class IidChangePointStaticExtensions
```

Comment:
>IidChangePointStaticExtensions [](start = 24, length = 30)

are you planning to add pigstensions? If not, this whole class is not needed

Reply:
removed for now

---
In reply to: [226105691](https://github.com/dotnet/machinelearning/pull/1254#discussion_r226105691) [](ancestors = 226105691)